src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -415,9 +433,12 @@ private static InetAddress getNetworkInterfaceAddress(String intf, String config
         }
     }
 
-    private static void setConfig(Config config)
+    private static void setConfig(Supplier<Config> config)
     {
-        conf = config;
+        conf = config.get();
+        confValueAccessors = ImmutableMap.copyOf(
```

Comment:
this is static, why are you setting it on `setConfig`?   `static final` is far easier to make sure safe than mutable

Reply:
Fixed.